lib/logdna/client.rb

```diff
@@ -63,7 +63,7 @@ def write_to_buffer(msg, opts)
         @lock.unlock
 
         flush if @flush_limit <= @buffer_byte_size
-        schedule_flush unless @flush_scheduled
+        schedule_flush
```

Comment:
I've lost context - I'm confused why we need this line (66) when we have line 65. Can you explain?

Reply:
There are two reasons why logs are flushed: either the buffer is full or the scheduled time is up. We can configure the library in such a way that it flushed, let's say, every 15 minutes or when there are more than x amount of data. If the size limit is 1000 byte, but we sent only 5 bytes, then it will wait for 15 minutes (or whatever time) to send the log.

Reply:
Then should there be some sort of `else` here? I'm confused why you would ever `flush` in line 65 and then immediate schedule another flush in line 66.

Reply:
yes, you are right.